src/libinjector/win_injector.c

```diff
@@ -741,6 +741,12 @@ static event_response_t wait_for_target_process_cb(drakvuf_t drakvuf, drakvuf_tr
             goto done;
         }
 
+        if (bp_addr & (1ULL << 63))
```

Comment:
How about `& VMI_BIT_MASK(47,63)`? Also, did you actually run into cases where the trapframe had a kernel address?

Reply:
look at https://github.com/tklengyel/drakvuf/issues/940

this happens sometimes for some threads and essentially causes deadlock because this points to either invalid place or some very rare code path

Reply:
Ough, yea, never thought that being possible. Very weird O.o

Reply:
hm, with all patches applied I can't reproduce these strange RIP values, maybe it's some other bug tricking us into this 🤔

Reply:
I wonder if what happened was we called set_cpuregs in one trapped vcpu callback just as the kernel was saving it on the other vcpu and we replaced the value with an old value. If we can't now reproduce it I wager that's what happened.